app/controllers/tasks_controller.rb

```diff
@@ -8,7 +8,7 @@ def show
 
   def update
     @task.actions.each do |action|
-      action_completed = action.id.in?(task_params[:completed_action_ids] || [])
+      action_completed = action.id.in?(params.dig("task", "actions") || [])
```

Comment:
This uses `params.dig` over `params[:task]` so that it neatly returns `nil` on failure rather than throwing an exception.

Reply:
I think we can take this refactor a little further, at the moment this is a bit of an odd mix, we are throwing arrays about here when I don't think we need to.

If we want to keep this `each` then `action.update(completed: action_completed)` might read better?

A further refinement might be something like:

`Action.update([array, of, all, task, ids], completed: [array, of, the, response, values])` 

to do everything in one DB call, which I think would be my preference.

We can also get the form builder to return both checked and uncheked states, which I think will make this even better as the unset boxes will also be in the params:

https://govuk-form-builder.netlify.app/form-elements/checkboxes/#generating-a-single-checkbox

Reply:
If we use the unchecked state as well, we can actually get the update down to an even cleaner one-line, since we can just use the value of the checkbox directly from the form and it'll be cast to a boolean (so `1`/`0` work perfectly).

I'm less certain about the single DB call, since that only works as long as everything is a checkbox so we're going to end up taking it back out again in future. I have, however, fixed that if a task is marked N/A it would be updating every checkbox with a DB hit and then unsetting them all again!